src/index.js

```diff
@@ -137,52 +104,68 @@ import sh from 'shelljs';
     await writeFile(mainModuleFilename, mainModuleContents, 'utf-8');
 
     // compile main module
-    const contents = await compileToString([mainModuleFilename], {
+    [elmCompileStdoutFd, elmCompileStderrFd] = await Promise.all([open(elmCompileStdout, 'w'), open(elmCompileStderr, 'w')]);
+
+    compileSync([mainModuleFilename], {
       yes: true,
-      report: reportFormat,
+      report,
       pathToMake: pathToElmMake,
+      output: outputCompiledFilename,
+      processOpts: {
+        stdio: [0, elmCompileStdoutFd, elmCompileStderrFd]
+      }
     });
-    await writeFile(outputCompiledFilename, contents);
+    if (!await exists(outputCompiledFilename)) {
+      const errorMessage = `Compilation failed\n${await readFile(elmCompileStderr, 'utf8')}${await readFile(elmCompileStdout, 'utf8')}`;
+      throw new Error(errorMessage);
+    }
 
     // run compiled elm file
+    const result = {
+      output: '',
+      debugLog: [],
+    };
+    const standardConsoleLog = console.log;
+    console.log = (...args) => result.debugLog.push(args.join(' '));
     await new Promise((resolve) => {
       const { worker } = require(outputCompiledFilename)[mainModule];
       const app = needArgs ? worker(argsToOutput) : worker();
       app.ports.sendOutput.subscribe((output) => {
-        console.log(output);
+        result.output = output;
         resolve();
       });
     });
+    console.log = standardConsoleLog;
+    return result;
   } catch (err) {
     // handle error
     let message;
     if (typeof err === 'object' && 'message' in err) {
       if (err.message.indexOf(`does not expose \`${outputName}\``) !== -1) {
         message = `Elm file \`${userModulePath}\` does not define \`${outputName}\`.`;
       } else if (err.message.indexOf(`I cannot find module '${userModule}'`) !== -1) {
-        message = `Elm file \`${userModulePath}\` cannot be reached from --project-dir \`${projectDir}\`. Have you configured \`source-directories\` in elm-package.json?`;
+        message = `Elm file \`${userModulePath}\` cannot be reached from project dir \`${resolvedProjectDir}\`. Have you configured \`source-directories\` in elm-package.json?`;
       } else {
         message = err.message;
       }
     } else {
       message = err;
     }
-    console.error('Error:', message);
-    if (err.stack && process.env.DEBUG) {
-      console.error(err.stack.substring(err.toString().length + 1));
-    }
-    exitCode = 1;
+    err.message = message && process.platform === 'win32'
+      ? message.replace(/\r\n/g, '\n')
+      : message;
+    throw err;
   } finally {
     // cleanup
+    await Promise.all([
+      close(elmCompileStdoutFd).catch(() => {}),
+      close(elmCompileStderrFd).catch(() => {}),
+    ]);
```

Comment:
Had to do this to support Windows, otherwise the files were left unclosed and could not be deleted